packages/signers/ledger-signer/src/ledger-signer.ts

```diff
@@ -133,18 +168,15 @@ export class LedgerSigner {
         Uint8Array.from(u16.enc(ss58Prefix ?? DEFAULT_SS58)),
       ),
     )
-    const res = Uint8Array.from(
-      await this.#safeSend(
-        CLA,
-        INS.getAddress,
-        seeAddressInDevice ? P1.showAddress : P1.getAddress,
-        P2,
-        bufToSend,
-      ),
-    )
-    const pubkey = res.slice(0, 32)
-    this.#pubkeys.set(getPubkeyMapKey(path1, path2), pubkey)
-    return pubkey
+    const prom = this.#safeSend(
+      CLA,
+      INS.getAddress,
+      seeAddressInDevice ? P1.showAddress : P1.getAddress,
+      P2[this.#schema],
+      bufToSend,
+    ).then((v) => Uint8Array.from(v).slice(0, -2)) // remove return code
+    this.#pubkeys.set(getPubkeyMapKey(this.#schema, path1, path2), prom)
```

Comment:
I think you already have the key from above?

```suggestion
    this.#pubkeys.set(key, prom)
```

Reply:
Yup, good catch. Indeed, the function can be directly inlined.